pkg/ingester/ingester.go

```diff
@@ -1006,6 +1006,7 @@ func (i *Ingester) Push(ctx context.Context, req *cortexpb.WriteRequest) (*corte
 			// of it, so that we can return it back to the distributor, which will return a
 			// 400 error to the client. The client (Prometheus) will not retry on 400, and
 			// we actually ingested all samples which haven't failed.
+			level.Warn(logutil.WithContext(ctx, i.logger)).Log("msg", "partial failure to push", "err", err)
```

Comment:
I am curious on if this log would be too noisy?

Reply:
I don't expect it to be. This is just in failures cases to append to tsdb, because of limit or storage issue. The idea behind is to use this with the new header to log feature (#4803). That would allow us to have a tracker between distributor and ingester issue.
Currently ingester does not log any message on this type of failures, and the log in the server middleware does not implement the header logs feature from cortex, which seems correct as it is a generic logging.